Shaders/PostProcess/TemporalAntiAliasing/private/ComputeTemporalAntiAliasing.fx

```diff
@@ -12,55 +14,246 @@ cbuffer cbCameraAttribs
     CameraAttribs g_PrevCamera;
 }
 
-Texture2D<float4> g_TextureCurrColor;
-Texture2D<float4> g_TexturePrevColor;
-Texture2D<float2> g_TextureMotion;
+cbuffer cbTemporalAntiAliasingAttribs
+{
+    TemporalAntiAliasingAttribs g_TAAAttribs;
+}
+
+Texture2D<float3> g_TextureCurrColor;
+Texture2D<float3> g_TexturePrevColor;
+Texture2D<float2> g_TextureCurrMotion;
+Texture2D<float2> g_TexturePrevMotion;
 Texture2D<float>  g_TextureDepth;
 
+SamplerState g_TexturePrevColor_sampler;
 
-float4 SampleCurrColor(int2 PixelCoord)
+struct PixelStatistic
 {
-    return g_TextureCurrColor.Load(int3(PixelCoord, 0));
+    float3 Mean;
+    float3 Variance;
+    float3 StdDev;
+};
+
+float3 RGBToYCoCg(float3 RGB)
+{
+    float3x3 Transform = float3x3(
+         0.25, 0.50,  0.25,
+         0.50, 0.00, -0.50,
+        -0.25, 0.50, -0.25
+    );
```

Comment:
In OpenGL, this is doing not what you expect as the matrix is transposed. You should use the `MatrixFromRows` function or simply compute the value directly without matrix multiplication